src/app/scenes/Category.tsx

```diff
@@ -1,56 +1,59 @@
 import React from 'react';
-import { connect } from 'react-redux';
-import { Link, LinkProps } from 'react-router-dom';
+import { useSelector, useDispatch } from 'react-redux';
+import { Link, LinkProps, useHistory } from 'react-router-dom';
 
 import { GridBookList, HelmetWithTitle, PCPageHeader } from 'app/components';
-import history from 'app/config/history';
 import { PageTitleText } from 'app/constants';
 import { GridBookListSkeleton } from 'app/placeholder/BookListPlaceholder';
-import { BookState } from 'app/services/book';
-import {
-  Category as CategoryState,
-  CategoryCollectionState,
-  Actions as categoryActions,
-} from 'app/services/category';
+import { Actions as categoryActions } from 'app/services/category';
 import { getIdFromLocationSearch, isValidNumber } from 'app/services/category/utils';
 import { RidiSelectState } from 'app/store';
 import { Pagination } from 'app/components/Pagination';
-import { getIsIosInApp } from 'app/services/environment/selectors';
 import { getPageQuery } from 'app/services/routing/selectors';
 import { getIsMobile } from 'app/services/commonUI/selectors';
 
-interface CategoryStateProps {
-  isIosInApp: boolean;
-  isCategoryListFetched: boolean;
-  categoryList: CategoryState[];
-  categoryId: number;
-  category: CategoryCollectionState;
-  books: BookState;
-  page: number;
-  isMobile: boolean;
-}
-
-type Props = CategoryStateProps & ReturnType<typeof mapDispatchToProps>;
-
-interface State {
-  isInitialized: boolean;
-}
-
-class Category extends React.Component<Props, State> {
-  private initialDispatchTimeout?: number | null;
-
-  public state: State = {
-    isInitialized: false,
-  };
+const Category: React.FunctionComponent = () => {
+  const dispatch = useDispatch();
+  const history = useHistory();
+
+  const isCategoryListFetched = useSelector((state: RidiSelectState) => state.categories.isFetched);
+  const categoryList = useSelector((state: RidiSelectState) => state.categories.itemList) || [];
+  const categoryId = useSelector((state: RidiSelectState) =>
+    Number(getIdFromLocationSearch(state.router.location.search)),
+  );
+  const category = useSelector(
+    (state: RidiSelectState) =>
+      state.categoriesById[Number(getIdFromLocationSearch(state.router.location.search))],
+  );
```

Comment:
```suggestion
  const category = useSelector((state: RidiSelectState) =>
    state.categoriesById[Number(getIdFromLocationSearch(state.router.location.search))],
  );
```

불필요한 줄넘김으로 보입니다. (바로 위의 `categoryId` 와 동일하게 수정)

Reply:
`Number(getIdFromLocationSearch(state.router.location.search)` 를 categoryId 로 대체할 방법은 없을까요?

Reply:
줄넘김은 prettier 자동수정 입니다.
categoryId 를 사용하도록 수정 했습니다.